### ISSUE TITLE
Implement InnerSpec Kotlin DSL

### ISSUE THREAD

**Issue by seongahjo** [naver/fixture-monkey#981]
## Describe the feature you request
InnerSpec now shadow the lambda parameter `it` if you use the nested InnerSpec.

```kotlin
class CustomObject(val value: String)

val actual = SUT.giveMeBuilder<Map<String, CustomObject>>()
          .setInner {
             it.size(1)
                    .entry("key") { v ->
                        v.property("value", "expected")
                    }
          }
          .sample()
```

You should specify the parameter as `v` if you want to customize the value of Map.
It is better to use Kotlin DSL for InnerSpec like below.

```kotlin
val actual = SUT.giveMeBuilder<Map<String, CustomObject>>()
          .setInner {
               size(1)
                entry("key") {
                        property("value", "expected")
                    }
          }
          .sample()
```

[InstantiationDsl](https://github.com/naver/fixture-monkey/blob/2e5d4f7bc7f183d6aa3ec1dce15cc83b11da46c5/fixture-monkey-kotlin/src/main/kotlin/com/navercorp/fixturemonkey/kotlin/instantiator/KotlinInstantiatorDsl.kt#L28-L28) is a good reference.

```kotlin
   val actual = SUT.giveMeBuilder<Foo>()
            .instantiateBy {
                constructor<Foo> {
                    parameter<String>()
                    parameter<Int>()
                }
            }
            .sample()
            .bar

        then(actual).isNotNull
```

**Comment by eunbileeme** [comment]
Can i contribute this?

**Comment by seongahjo** [comment]
@eunbileeme 
Sure!  Let me know if you have any questions.

**Comment by seongahjo** [comment]
@eunbileeme 
Hello, are there any updates?

**Comment by eunbileeme** [comment]
Hello.

Sorry, I was already mistaken that someone else solved it.

Is there a set period for the issue to close?

**Comment by seongahjo** [comment]
@eunbileeme 
 No, no fixed deadline. But let me know if you do not have time to resolve this issue for other contributors. 😄

**Comment by eunbileeme** [comment]
@seongahjo 
Checked it.
I will commit as soon as possible or share the schedule.

Once again, I'm sorry for the late confirmation. :

**Comment by soonhong99** [comment]
@eunbileeme @seongahjo 

any updates?

**Comment by seongahjo** [comment]
@soonhong99 
Sadly, it seems there is no update for now.

Perhaps someone else can resolve with this.

**Comment by soonhong99** [comment]
> @soonhong99 Sadly, it seems there is no update for now.
> 
> Perhaps someone else can resolve with this.

Then can I resolve this issue?

**Comment by seongahjo** [comment]
@soonhong99 
Sure, thank you!

**Comment by seongahjo** [comment]
Hello, @kssumin 

> I'd like to implement the InnerSpec Kotlin DSL solution for issue

Good to hear that!  I think your plan sounds reasonable. Please go ahead!

Thank you.
